pkg/handler/collector/git/git_test.go

```diff
@@ -0,0 +1,170 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git_collector
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+func Test_gitCol_RetrieveArtifacts(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
+	logger := logging.FromContext(ctx)
+	type fields struct {
+		url      string
+		dir      string
+		poll     bool
+		interval time.Duration
+	}
+	tests := []struct {
+		name                   string
+		fields                 fields
+		preCreateDir           bool
+		wantDirEmpty           bool
+		wantErr                bool
+		numberOfFilesCollected int
+	}{{
+		name: "get repo",
+		fields: fields{
+			url:      "https://github.com/shafeeshafee/test-123.git",
+			dir:      "./guac-data-test",
+			poll:     false,
+			interval: time.Millisecond,
+		},
+		preCreateDir:           false,
+		wantDirEmpty:           false,
+		wantErr:                false,
+		numberOfFilesCollected: 9,
+	}, {
+		name: "if repo exist",
+		fields: fields{
+			url:      "https://github.com/shafeeshafee/test-123.git",
+			dir:      "./guac-data-test",
+			poll:     false,
+			interval: time.Millisecond,
+		},
+		preCreateDir:           true,
+		wantDirEmpty:           false,
+		wantErr:                false,
+		numberOfFilesCollected: 0,
+	}, {
+		name: "get repo poll",
+		fields: fields{
+			url:      "https://github.com/shafeeshafee/test-123.git",
+			dir:      "./guac-data-test",
+			poll:     true,
+			interval: time.Millisecond,
+		},
+		preCreateDir:           false,
+		wantDirEmpty:           false,
+		wantErr:                false,
+		numberOfFilesCollected: 9,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// incase the file exists from a failed run, delete it
+			os.RemoveAll(tt.fields.dir)
+			g := NewGitCol(ctx, tt.fields.url, tt.fields.dir, tt.fields.poll, tt.fields.interval)
+
+			if tt.preCreateDir {
+				if err := os.Mkdir(tt.fields.dir, os.ModePerm); err != nil {
+					t.Fatal(err)
+				}
+				err := cloneRepoToTemp(logger, tt.fields.url, tt.fields.dir)
+				if err != nil {
+					t.Fatal(err)
+				}
+			}
+
+			var cancel context.CancelFunc
+			if tt.fields.poll {
+				ctx, cancel = context.WithTimeout(ctx, time.Second)
+				defer cancel()
+			}
+
+			docChan := make(chan *processor.Document, 1)
+			defer os.RemoveAll(tt.fields.dir) // clean up
+			errChan := make(chan error, 1)
+			defer close(docChan)
+			defer close(errChan)
+			go func() {
+				errChan <- g.RetrieveArtifacts(ctx, docChan)
+			}()
+
+			numCollectors := 1
+			collectorsDone := 0
+
```

Comment:
hm, might need some clarification for this. Are you suggesting that we call the collector.Collect function for the unit test?

Reply:
hmmm yea i think that would work... don't think it will introduce a circular testing dependency. Could be a different PR to do it for the other collectors as well though!

Reply:
yea maybe as a separate PR, i think there are things we can do to refactor that to make it nicer for test cases... if not its really messy with the registration...

Reply:
Okay ^, created a new issue based on this, will update some of its details a bit later. 

https://github.com/guacsec/guac/issues/281